### ISSUE TITLE
auto_zoom_on_focus doesn't play nice with widget plugins

### ISSUE THREAD

**Issue by dyereh** [SublimeText/Origami#148]
Hi - I really like the auto_zoom_on_focus feature but have found that it doesn't work with widget plugins such as terminus. I believe it is due to this bit of code:

```
if view.settings().get("is_widget"):
	return
```

Any way we this feature could be made to work in widgets such as terminus?

**Comment by adzenith** [comment]
Just to clarify - you're saying that you want terminus to resize when it gets focus? I'm not sure if the API allows resizing widgets.

**Comment by dyereh** [comment]
Yup that's it - will see if I can get it to work and let ya know

**Comment by dyereh** [comment]
Yup just tested - by deleting those two lines terminus will also resize. I'm curious why these were there to begin with. Did you not want widgets resizing?

This is a fantastic feature particularly when I switch to a single monitor and don't have as much real-estate but don't wanna change my pane layout

**Comment by dyereh** [comment]
I also found it useful to be able to toggle the auto_zoom_on_focus feature on and off from a keybinding. Not sure if this is a useful thing to include in the plugin. Right now I just have a little bit of code to wrap it

`  { "keys": ["Z"], "command": "toggle_plugin_setting", "args": {"file": "Origami.sublime-settings", "key": "auto_zoom_on_focus", "value1": false, "value2":0.5 }, "context": [{"key": "get_cmd_mode", "operand": true}] },
`

`import sublime, sublime_plugin

## Toggle a setting for any plugin
class TogglePluginSetting(sublime_plugin.ApplicationCommand):
    toggle = 0

    def run(self, file, key, value1, value2):
        print(self.toggle)
        settings = sublime.load_settings(file)

        if self.toggle == 0:
            value = value1
        else:
            value = value2

        settings.set(key, value)
        sublime.save_settings(file)

        self.toggle = (self.toggle + 1) % 2`

I could PR this in at some point if you found it useful..

**Comment by juliangarnier** [comment]
+1, being able to resize the pane containing a terminus instance would be great.
